src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift

```diff
@@ -47,70 +48,22 @@ final class ExposureDetection {
 		progress?.cancel()
 	}
 
-	private func getSupportedCountries(completion: @escaping ([Country]) -> Void) {
-		delegate?.exposureDetection(supportedCountries: { [weak self] result in
-			guard let self = self else { return }
-
+	private func downloadKeyPackages(completion: @escaping () -> Void) {
+		self.countryKeypackageDownloader.downloadKeypackages(for: country) { result in
 			switch result {
-			case .success(let supportedCountries):
-				var _supportedCountries = supportedCountries
-
-				// If supported countries is empty for some reason, we add the default country (DE).
-				if _supportedCountries.isEmpty {
-					_supportedCountries.append(Country.defaultCountry())
-				}
-
-				completion(_supportedCountries)
-			case.failure:
-				self.endPrematurely(reason: .noSupportedCountries)
-			}
-		})
-	}
-
-	private func downloadKeyPackages(for countries: [Country.ID], completion: @escaping () -> Void) {
-
-		let dispatchGroup = DispatchGroup()
-		var errors = [ExposureDetection.DidEndPrematurelyReason]()
-
-		for country in countries {
-			dispatchGroup.enter()
-
-			self.countryKeypackageDownloader.downloadKeypackages(for: country) { result in
-				switch result {
-				case .failure(let didEndPrematurelyReason):
-					errors.append(didEndPrematurelyReason)
-				case .success:
-					break
-				}
-
-				dispatchGroup.leave()
-			}
-		}
-
-		dispatchGroup.notify(queue: .main) {
-			if let error = errors.first {
-				self.endPrematurely(reason: error)
-			} else {
+			case .failure(let didEndPrematurelyReason):
+				self.endPrematurely(reason: didEndPrematurelyReason)
```

Comment:
like above:

```suggestion
				endPrematurely(reason: didEndPrematurelyReason)
```

Reply:
This one is inside of a closure. But thanks for pointing this our. I have made self weak.